src/Optical/Lenses.jl

```diff
@@ -2,123 +2,376 @@
+        interfacemode
+    )
 end
-export ConicLens
 
 """
-    AsphericLens(insidematerial, frontvertex, frontradius, frontconic, frontaspherics, backradius, backconic, backaspherics, thickness, semidiameter;  lastmaterial = OpticSim.GlassCat.Air, nextmaterial = OpticSim.GlassCat.Air, frontsurfacereflectance = 0.0, backsurfacereflectance = 0.0, frontdecenter = (0, 0), backdecenter = (0, 0), interfacemode = ReflectOrTransmit)
+    AsphericLens(
+        insidematerial,
+        frontvertex,
+        frontradius,
+        frontconic,
+        frontaspherics,
+        backradius,
+        backconic,
+        backaspherics,
+        thickness,
+        semidiameter;
+        lastmaterial = Air,
+        nextmaterial = Air,
+        frontsurfacereflectance = 0.0,
+        backsurfacereflectance = 0.0,
+        frontdecenter = (0, 0),
+        backdecenter = (0, 0),
+        interfacemode = ReflectOrTransmit
+    )
 
-Cosntructs a simple cylindrical lens with front and back surfaces with a radius, conic and apsheric terms. The side walls of the lens are absorbing.
+Cosntructs a simple cylindrical lens with front and back surfaces with a radius, conic and apsheric terms. The side
+walls of the lens are absorbing.
 """
-function AsphericLens(insidematerial::T, frontvertex::S, frontradius::S, frontconic::S, frontaspherics::Union{Nothing,Vector{Tuple{Int,S}}}, backradius::S, backconic::S, backaspherics::Union{Nothing,Vector{Tuple{Int,S}}}, thickness::S, semidiameter::S; lastmaterial::Q = OpticSim.GlassCat.Air, nextmaterial::R = OpticSim.GlassCat.Air, frontsurfacereflectance::S = zero(S), backsurfacereflectance::S = zero(S), nsamples::Int = 17, frontdecenter::Tuple{S,S} = (zero(S), zero(S)), backdecenter::Tuple{S,S} = (zero(S), zero(S)), interfacemode = ReflectOrTransmit) where {R<:OpticSim.GlassCat.AbstractGlass,Q<:OpticSim.GlassCat.AbstractGlass,T<:OpticSim.GlassCat.AbstractGlass,S<:Real}
-    @assert semidiameter > zero(S)
+function AsphericLens(
+    insidematerial::AbstractGlass,
+    frontvertex::T,
+    frontradius::T,
+    frontconic::T,
+    frontaspherics::Union{Nothing,Vector{Tuple{Int,T}}},
+    backradius::T,
+    backconic::T,
+    backaspherics::Union{Nothing,Vector{Tuple{Int,T}}},
+    thickness::T,
+    semidiameter::T;
+    lastmaterial::AbstractGlass = Air,
+    nextmaterial::AbstractGlass = Air,
+    frontsurfacereflectance::T = zero(T),
+    backsurfacereflectance::T = zero(T),
+    nsamples::Int = 17,
+    frontdecenter::Tuple{T,T} = (zero(T), zero(T)),
+    backdecenter::Tuple{T,T} = (zero(T), zero(T)),
+    interfacemode::InterfaceMode = ReflectOrTransmit
+) where {T<:Real}
+    @assert semidiameter > zero(T)
     @assert !isnan(frontradius)
 
-    frontdecenter_l = abs(frontdecenter[1]) > eps(S) && abs(frontdecenter[2]) > eps(S) ? sqrt(frontdecenter[1]^2 + frontdecenter[2]^2) : zero(S)
-    backdecenter_l = abs(backdecenter[1]) > eps(S) && abs(backdecenter[2]) > eps(S) ? sqrt(backdecenter[1]^2 + backdecenter[2]^2) : zero(S)
```

Comment:
Is it possible to do `x != zero(S)` instead of `abs(x) > eps(S)` here? Not sure if there's a particular use case of decentering that we are accounting for.

Reply:
I think eps was used to allow for some numeric error as the decenter values are calculated, not sure if there is much practical impact to use zero instead - maybe add a test for this edge case?